.github/workflows/python-ci.yml

```
@@ -410,7 +410,7 @@ jobs:
     name: SonarCloud Quality Gate
     runs-on: ubuntu-latest
     needs: [quality-checks]
-    if: inputs.enable-sonarcloud
+    if: inputs.enable-sonarcloud && secrets.SONAR_TOKEN != ''
```

actionlint (1.7.9)

context "secrets" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)